nomenclature/codes.py

```diff
@@ -105,14 +104,14 @@ def from_directory(
             # validate against the schema of this codelist domain
             validate(_code_list, SCHEMA_MAPPING[name])
 
-            # if specified, set top-level key as attribute instead
-            if top_level_attr is not None:
+            # a "region" codelist assumes a top-level key to be used as attribute
+            if name == "region":
                 original_code_list, _code_list = _code_list.copy(), []
```

Comment:
`_code_list.copy()` creates a shallow copy. Is that fine or do we actually want a deep copy here?

Reply:
Refactored the copy to avoid the copy alltogether...